config/nrfconnect/chip-module/CMakeLists.txt

```diff
@@ -139,6 +139,12 @@ matter_add_gn_arg_bool  ("chip_enable_icd_server"                 CONFIG_CHIP_EN
 matter_add_gn_arg_bool  ("chip_enable_factory_data"               CONFIG_CHIP_FACTORY_DATA)
 matter_add_gn_arg_bool  ("chip_enable_read_client"                CONFIG_CHIP_ENABLE_READ_CLIENT)
 
+if (CONFIG_CHIP_ENABLE_ICD_SUPPORT)
+    matter_add_gn_arg_bool  ("chip_enable_icd_lit"                 CONFIG_CHIP_ICD_LIT_SUPPORT)
+    matter_add_gn_arg_bool  ("chip_enable_icd_cip"                 CONFIG_CHIP_ICD_CHECK_IN_SUPPORT)
+    matter_add_gn_arg_bool  ("chip_enable_icd_uat"                 CONFIG_CHIP_ICD_UAT_SUPPORT)
```

Comment:
my understanding from the spec is that cip and uat are required if lit support is enabled. do we need need 3 defines or can we live only with lit enable/disable?

Reply:
a SIT can enable the features without the lit icd flag thus the brake out.

Reply:
Yes, in fact `chip_enable_icd_uat` flag is not used anywhere yet, as ICD core does not seems to handle UAT there, but I thought it could be useful for an application purposes. This way it would be possible to create a code in the application that updates operating mode if `CHIP_CONFIG_ENABLE_ICD_UAT` and then select the compiled implementation based on the gn flag.